src/grappa_interface.py

```diff
@@ -25,12 +24,12 @@ def __init__(
         super().__init__(*args, **kwargs)
         grappa_instance = Grappa.from_tag(grappa_tag)
         self.kimmdy_grappa_parameterizer = KimmdyGrappaParameterizer(
-            grappa_instance=grappa_instance, charge_model=charge_model
+            grappa_instance=grappa_instance, charge_model=charge_model, grappa_tag = grappa_tag
         )
 
     def parameterize_topology(
-        self, current_topology: Topology, focus_nr: list[str] = []
+        self, current_topology: Topology, focus_nrs: set = set()
```

Comment:
This can lead to a subtle bug that is hard to debug. Setting the default argument to a mutable container initializes it at the time of definition, which means all users of the function will actually share the same container instead of getting their own empty set. The canonical workaround for this is to set the default to `None` and create the empty set on None-check in the body of the function.